kld/src/ldk/event_handler.rs

```diff
@@ -150,7 +154,19 @@ impl EventHandler {
                     "EVENT: Channel {} - {user_channel_id} with counterparty {counterparty_node_id} is ready to use.",
                     channel_id.encode_hex::<String>(),
                 );
-                self.peer_manager.broadcast_node_announcement();
+                let mut alias = [0; 32];
+                alias[..self.settings.node_alias.len()]
+                    .copy_from_slice(self.settings.node_alias.as_bytes());
+                let addresses: Vec<lightning::ln::msgs::NetAddress> = self
+                    .settings
+                    .public_addresses
+                    .clone()
+                    .into_iter()
+                    .map(|a| a.inner())
+                    .collect();
+                info!("Broadcasting node announcement message");
+                self.peer_manager
+                    .broadcast_node_announcement([0; 3], alias, addresses);
```

Comment:
Don't know why this code is copied here. Just leave it as a function call.

Reply:
This code is not a copy.  There is no duplicated PeerManager to store `settings.node_alias`, `settings.public_address`. The logic to extract `node_alias` and `public_address` move here.

The [`broadcast_node_announcement`](https://docs.rs/lightning/0.0.116/lightning/ln/peer_handler/struct.PeerManager.html#method.broadcast_node_announcement) is defined and exposed in `lightning`.
It is confusing to hide a lib function by making a same-name function without inputs.

If these logics should be in `peer_manager` and not be `event_handler`, the `broadcast_node_announcement_from_setting` is created now to encapsulate these.